lib/async_hooks.js

```diff
@@ -211,6 +211,14 @@ class AsyncResource {
   triggerAsyncId() {
     return this[trigger_async_id_symbol];
   }
+
+  bind(fn) {
+    return this.runInAsyncScope.bind(this, fn);
+  }
+
+  static bind(fn, type = fn?.name) {
+    return (new AsyncResource(type || 'bound-anonymous-fn')).bind(fn);
```

Comment:
no points will be deducted for using a temporary variable :P

Reply:
parentheses unnecessary?

Reply:
Yeah but stylistically I prefer them (more readable in my opinion)